addons/event_booth_sale/wizard/event_booth_batch_configurator_line.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class EventConfiguratorLine(models.TransientModel):
+    """Event Booth Configuration"""
+    _name = "event.booth.editor.line"
+    _description = 'Edit Event Booth values on Sales Confirmation'
+    _order = "id desc"
+
+    editor_id = fields.Many2one('event.event.booth.configurator.batch')
+    product_id = fields.Many2one('product.product', string="Product", readonly=True)
+    event_booth_category_id = fields.Many2one(
+        'event.booth.category', string='Booth Category', required=True,
+        compute='_compute_event_booth_category_id', readonly=False, store=True)
+    event_booth_ids = fields.Many2many(
+        'event.booth', string='Booth', required=True,
+        compute='_compute_event_booth_ids', readonly=False, store=True)
+    event_id = fields.Many2one('event.event', string='Event', required=True)
+    event_booth_category_available_ids = fields.Many2many(related='event_id.event_booth_category_available_ids', readonly=True)
+    order_id = fields.Many2many('sale.order', string="Sale order")
+    sale_order_line_id = fields.Many2many('sale.order.line', string="Sale Order Line")
+
+    @api.depends('event_id')
+    def _compute_event_booth_category_id(self):
+        self.event_booth_category_id = False
+
+    @api.depends('event_id', 'event_booth_category_id')
+    def _compute_event_booth_ids(self):
+        self.event_booth_ids = False
+
+    def get_event_data(self):
+        self.ensure_one()
+        for line in self.sale_order_line_id:
+            if line.product_type == 'event_booth' and not line.event_id.id:
+                line.write({
+                    'event_id': self.event_id.id,
+                    'order_id': self.order_id.id,
+                    'event_booth_ids': self.event_booth_ids.ids,
+                    'event_booth_category_id': self.event_booth_category_id.id,
```

Comment:
this is not extensible, you overwrite the entire method to write 2 other field, "event_booth_ids" and "event_booth_category_id", but what if one day we have a different module depending on event_sale but not on event_booth_sale ?

one module will overwrite the other and some field will be missing

instead what you can do is a prepare method that return the value to write, and you call the super and concatenate the "super values" with the current one

something like
```python
def _prepare_event_values(self):
    return {
        ** super()._prepare_event_values()
        "event_booth_ids": ...,
        "event_booth_category_id": ...,
    }
```

Reply:
Hey, thanks for review! I have a small question regarding this part:

Since we are creating a new model for the wizard and not inheriting one, it will not override anything even if we have similar function names no? Thus, if someone will want their module to depend on event_sale and not on event_sale_booth, they will be changing _get_event_data function of "event.editor.line" model, not the _get_event_data function of "event.booth.editor.line" model. Am I missing something in the logic here?

Reply:
ah yes sorry, you are right :) mistake from me, sorry